lib/spack/spack/spec.py

```diff
@@ -2221,24 +2234,23 @@ def normalize(self, force=False, tests=False):
         # Get all the dependencies into one DependencyMap
         spec_deps = self.flat_dependencies(copy=False)
 
+        full_spec_deps = dict(spec_deps)
```

Comment:
Why create a new `dict`?  At this point you should just update `spec_deps` and not bother with another `dict`.  It probably needs another name now that it includes more.